pkg/controller/direct/secretmanager/secret_controller.go

```diff
@@ -262,6 +239,10 @@ func (a *Adapter) Update(ctx context.Context, op *directbase.UpdateOperation) er
 	if mapCtx.Err() != nil {
 		return mapCtx.Err()
 	}
+	// Set externalRef. This field is empty after migration from the TF-based controller to the direct.
```

Comment:
Secret is not service generated ID resource (SecretVersion does). So it can switch back to use the TF-based controller. The real problem on using the TF-based controller is that the Secret `spec.rotation.rotationPeriod` is [broken](https://screenshot.googleplex.com/7W8MxZsxeE527sV.png). If user configures that fields in Direct and switched back to TF-based, it no longer works.